homeassistant/components/iss/binary_sensor.py

```diff
@@ -78,6 +79,11 @@ def device_class(self):
         """Return the class of this sensor."""
         return DEFAULT_DEVICE_CLASS
 
+    @property
+    def icon(self):
+        """Return the icon of this sensor."""
+        return self._icon
```

Comment:
This entity has a device class. We either set a device class or an icon on an entity, not both.

Reply:
I thought that device class is just representation of the status - 'visible' or not instead of just 'on' or 'off'. it has nothing to do with the icon.. what am I missing? I know it can be configured using customize, but as a default it is more intuitive..
and this rule does not appear anywhere in the design guidlines, and a lot of integrations providing both device class and icons..

Reply:
If a device class is set, the handling of the default icon is done by the UI, not the backend.
Hence you set either the one or the other.

Reply:
Interesting - but when the device class is not from the list like we have here, the UI does not have a matching icon for it and it shows the default boolean icon..
So what do you suggest? close this PR or remove the device class (which is totaly legit to keep as the sensor is 'visible' and not 'on')?

Reply:
> Interesting - but when the device class is not from the list like we have here, the UI does not have a matching icon for it and it shows the default boolean icon..
> So what do you suggest? close this PR or remove the device class (which is totaly legit to keep as the sensor is 'visible' and not 'on')?

When the sensor is 'on' does it show `visible`'?  I just see 'on' on the frontend.

Reply:
> So what do you suggest? close this PR or remove the device class

In general, we prefer device classes. This is about representation in the frontend, which should not be a backend concern (at least, reducing that as much as possible).

Reply:
Thank you for the review and explenation @frenck  👍

Reply:
@frenck, I'm just curious after looking at this code, is it even possible to have a custom device class like this integration is using? 

It uses `DEFAULT_DEVICE_CLASS = "visible"`. From what I can tell that doesn't do anything does it?